### ISSUE TITLE
Selecting records with no associated record using many to many association

### ISSUE THREAD

**Issue by damien-git** [sequelize/sequelize#12117]
##  Issue Description
The inclusion of the through attributes in the query prevents the use of GROUP BY. I understand the below code doesn't work because `attributes:[]` is only filtering attributes after the query is executed, but I am not seeing any workaround when trying to get records with a count of associated records equal to 0.
This is a simplified case. My real query actually includes more tables and conditions.
I think this issue is related to #11819.

### What are you doing?

```js
const users = await User.findAll({
  attributes: ['id'],
  include: [
    {
      model: Unit,
      attributes: [],
      through: {
        attributes: [],
      },
    },
  ],
  group: ['id'],
  having: Sequelize.where(Sequelize.fn('COUNT', Sequelize.col('units.id')), 0),
  raw: true,
});

```

### What do you expect to happen?

I would like the through attributes to not appear in the select clause of the query, so that the GROUP BY works.

### What is actually happening?

``
SELECT `user`.`id`, `units->userUnit`.`role` AS `units.userUnit.role`, `units->userUnit`.`unit_id` AS `units.userUnit.unitId`, `units->userUnit`.`user_id` AS `units.userUnit.userId` FROM `users` AS `user` LEFT OUTER JOIN ( `users_units` AS `units->userUnit` INNER JOIN `units` AS `units` ON `units`.`id` = `units->userUnit`.`unit_id`) ON `user`.`id` = `units->userUnit`.`user_id` GROUP BY `user`.`id` HAVING COUNT(`units`.`id`) = 0;
``
``
UnhandledPromiseRejectionWarning: SequelizeDatabaseError: Expression #2 of SELECT list is not in GROUP BY clause and contains nonaggregated column 'override_requests.units->userUnit.role' which is not functionally dependent on columns in GROUP BY clause; this is incompatible with sql_mode=only_full_group_by
``

### Environment

- Sequelize version: 5.21.5
- Node.js version: 12.13.0
- Operating System: Debian Buster

### How does this problem relate to dialects?

- [ ] I think this problem happens regardless of the dialect.
- [ ] I think this problem happens only for the following dialect(s): 
- [X] I don't know, I was using mysql, with connector library version mysql2 2.1.0 and database version 8.0

### Would you be willing to resolve this issue by submitting a Pull Request?

- [ ] Yes, I have the time and I know how to start.
- [ ] Yes, I have the time but I don't know how to start, I would need guidance.
- [ ] No, I don't have the time, although I believe I could do it if I had the time...
- [X] No, I don't have the time and I wouldn't even know how to start.

**Comment by sushantdhiman** [comment]
Please search old issue before reporting new one, thanks.

https://github.com/sequelize/sequelize/issues/7198#issuecomment-284137408

**Comment by damien-git** [comment]
You might have noticed I linked to a previous issue, showing that I did search old issues before taking the time to report a new one. I have in fact looked at issues for over an hour, and seen several related issues (including the one you linked to), but none provide a solution.
The comment you link to suggests not using `raw:true`. That does not resolve the issue, which is (as describe above) that the inclusion of the through attributes in the `select` part of the query prevents the use of `group by`. Without `raw:true`, the generated SQL is exactly the same, so it fails too.
